sqlglot/parser.py

```diff
@@ -6485,10 +6491,14 @@ def _parse_alter_table_set(self) -> exp.AlterSet:
 
         return alter_set
 
+    def _parse_alter_view_as(self) -> t.Optional[exp.Expression]:
+        return self._parse_select(self._prev)
```

Comment:
Why are we passing `self._prev` to `_parse_select` here? We can also just move this inline within the `ALTER_PARSERS` dict, like we do for `"DELETE"`.

Reply:
fixed